lambda/es-proxy-layer/lib/esbodybuilder.js

```diff
@@ -104,11 +104,40 @@ function build_query(params) {
       if (_.get(params, 'score_answer_field')) {
         query = query.orQuery('match', 'a', params.question);
       }
-      query = query.orQuery('match', 't', _.get(params, 'topic', ''))
+      let topic = _.get(params, 'topic');
+      if (topic) {
+        query = query.orQuery('match', 't', topic);
+      } else {
+        // no topic - query prefers answers with empty/missing topic field for predicable response
+        // NOTE: will not work in Kendra FAQ mode since we have no equivalent Kendra query
+        query = query.orQuery(
+          'bool', {
+            "should" : [
```

Comment:
Could you please provide steps to tests this feature (test case with required assets (e.g., sample inputs)), so our QA can added to our test cases for the coming release.

Reply:
Sure..
In Content Designer, load example package "blog-samples-final"
There will be 2 separate QIDs that can answer the question "How much does it cost"  (see our documentation / blog example for 'answering follow-up questions using topics". One of these is for topic "QnABot', the other is for topic "EchoShow".
Use Content Designer to create a 3rd QID with the exact same question, different answer, and do not assign a value to the topic field.
Now send the question "How much does it cost?" to the bot using the web client, or the Content designer TEST tab.
Without this fix, the response from the bot could be unpredictable - could be any of the 3 QIDs that match the question, since there is no topic value set.
With this fix, the answer will always be the new Qid just added that has no topic value defined.